src/Microsoft.AspNetCore.Razor.Tools/DiscoverCommand.cs

```diff
@@ -150,15 +150,15 @@ private int ExecuteCore(RazorConfiguration configuration, string projectDirector
             {
                 Serialize(stream, tagHelpers);
 
-                stream.Position = 0L;
+                stream.Position = 0;
 
                 var newHash = Hash(stream);
                 var existingHash = Hash(outputFilePath);
 
                 if (!HashesEqual(newHash, existingHash))
                 {
                     stream.Position = 0;
-                    using (var output = File.OpenWrite(outputFilePath))
+                    using (var output = File.Open(outputFilePath, FileMode.Create))
```

Comment:
OpenWrite does not reset the file length.

Reply:
Oh. So does that mean that we could corrupt the contents?

Reply:
Yup. It works fine as the content was the same length or longer.